docs/pipelineruns.md

```diff
@@ -286,9 +286,47 @@ spec:
 
 ## Workspaces
 
-It is not yet possible to specify [workspaces](tasks.md#workspaces) via `Pipelines`
-or `PipelineRuns`, so `Tasks` requiring `workspaces` cannot be used with them until
-[#1438](https://github.com/tektoncd/pipeline/issues/1438) is completed.
+Workspaces allow PVCs and emptyDir volume sources to be
+easily wired into tasks and pipelines.
+
+For a `PipelineRun` to execute [a Pipeline that declares `workspaces`](pipelines.md#workspaces),
+at runtime you need to map the `workspace` names to actual physical volumes.
+This is managed through the `PipelineRun`'s `workspaces` field. Values in `workspaces` are
+[`Volumes`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-volume-storage/),
+however currently we only support a subset of `VolumeSources`:
+
+_If you need support for a `VolumeSource` not listed here
+[please open an issue](https://github.com/tektoncd/pipeline/issues) or feel free to
+[contribute a PR](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)._
+
+* [`emptyDir`](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)
+* [`persistentVolumeClaim`](https://kubernetes.io/docs/concepts/storage/volumes/#persistentvolumeclaim)
+* [`configMap`](https://kubernetes.io/docs/concepts/storage/volumes/#configmap)
+
+If the `workspaces` declared in the Pipeline are not provided at runtime, the `PipelineRun` will fail
+with an error.
+
+For example to provide an existing PVC called `mypvc` for a `workspace` called
+`myworkspace` declared by the `Pipeline`, using the `my-subdir` folder which already exists
+on the PVC (there will be an error if it does not exist):
+
+```yaml
+workspaces:
+- name: myworkspace
+  persistentVolumeClaim:
+    claimName: mypvc
+  subPath: my-subdir
+```
+
+Or to use [`emptyDir`](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for the same `workspace`:
```

Comment:
can emptyDir be passed across Tasks in a Pipeline? I was under the impression they are bound to a task/pod?

Reply:
You're right - they don't get passed across.  I'll clarify this in the docs.

Reply:
configmap looks good as an alternative for folks that don't have a PVC available but i think there is limit to how much we can store in a configmap, we may want to mention it? It seems that etcd limits it to 1MB ie: (old issue tho) https://github.com/kubernetes/kubernetes/issues/19781

Reply:
have updated docs with mention of emptyDir being newly created for each task & linked to kubernetes code showing the MaxSecretSize check (used for both configmap and secrets).  I couldn't find this max size documented anywhere in k8s' docs unfortunately.